lib/processEventDataHelper.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+const moment = require('moment-timezone');
+const _ = require('lodash');
+const Q = require('q');
+
+function checkRequiredFields(cfg, messageBody) {
+  console.log('Checking required fields ...');
+  if (!messageBody.start) {
+    throw new Error('Start Time missing! This field is required!');
+  }
+  if (!messageBody.end) {
+    throw new Error('End Time missing! This field is required!');
+  }
+  if (!cfg.timeZone) {
+    throw new Error('Time Zone missing! This field is required!');
+  }
+  if (!cfg.calendarId) {
+    throw new Error('Calendar ID missing! This field is required!');
+  }
+  if ((cfg.bodyContentType) && (!messageBody.body)) {
+    throw new Error('Body Type provided, but Body Content is missing!');
+  }
+}
+
+
+function formatDate(date, timeZone, format) {
```

Comment:
This seems to be the one of the most critical pieces of the component. Please write isolated tests for this function where you invoke ``formatDate()`` directly in the test. The test should cover all the supported date formats.